lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2022 Dynatrace LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.file.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Logger;
+
+class PollBasedFilePoller extends AbstractFilePoller {
+  private final AtomicBoolean changedSinceLastPoll = new AtomicBoolean(false);
+  private static final Logger logger = Logger.getLogger(PollBasedFilePoller.class.getName());
+  private volatile long lastUpdatedAt = 0;
```

Comment:
```suggestion
  private volatile long lastUpdatedAtMillis = 0;
```
Or directly store the `FileTime`?

Reply:
Either adapting the var name to indicate what the long is about or using FileTime directly is fine, it's up to you.

Reply:
Moved it back to millis